src/workloads/execution/SecondaryReadsGenny.yml

```diff
@@ -0,0 +1,231 @@
+MaxIdForFinds: &max_id_for_finds 9900
+TestDurationInSecs: &phase_duration 180 seconds
+Db: &db test
+WriterThreads: &writer_threads 16
+NumPhases: &max_phase 13
+NumberOfNodes: &number_of_nodes 3
+
+# This client allows the ReadsOnSecondary actor to directrly connect to the secondary node.
+# All other actors use the default client to connect to the primary node.
+Clients:
+  Secondary:
+    URI: mongodb://md1:27017
+
+ActorTemplates:
+- TemplateName: WritesOnPrimaryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: WritesOnPrimary}}
+    Type: CrudActor
+    Threads: *writer_threads
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Duration: *phase_duration
+          Collection: Collection0
+          Operations:
+          - OperationName: insertOne
+            OperationCommand:
+              Document:
+                x: {^RandomInt: {min: *initial_nb_docs, max: 1000000000}}
+                s: {^RandomString: {length: *document_size}}
+
+- TemplateName: ReadsOnSecondaryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: ReadsOnSecondary}}
+    Type: CrudActor
+    Threads: {^Parameter: {Name: "Threads", Default: -1}}
+    ClientName: Secondary
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Duration: *phase_duration
+          Collection: Collection0
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: { x: {$gte: {^RandomInt: {min: 0, max: *max_id_for_finds}}}}
+              Options:
+                Limit: *read_limit
+                BatchSize: 200
+                ReadPreference:
+                  ReadMode: secondaryPreferred
+
+- TemplateName: ReplicationLagTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: ReplicationLag}}
+    Type: CrudActor
+    Threads: 1
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Repeat: 1
+          Collection: Collection0
+          Operations:
+          - OperationName: insertOne
+            OperationCommand:
+              Document: {s: 1}
+              OperationOptions:
+                WriteConcern:
+                  Level: *number_of_nodes
+                  Journal: true
+
+
+Actors:
+- Name: LoadInitialData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0, 5, 10]
+      NopInPhasesUpTo: *max_phase
+      PhaseConfig:
+        Repeat: 1
+        Database: *db
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *initial_nb_docs
+        BatchSize: *initial_nb_docs
+        Document:
```

Comment:
I'm not the most familiar here so apologies if this is done elsewhere, should these commands be done with writeConcern equal to the number of nodes, so that we guarantee they land on all secondaries before reading?

Reply:
Good catch, it's important to ensure that the documents are propagated to all nodes before we start the test. We currently do this via the Quiesce phase after loading the initial data, but wc:3 would also work.

Reply:
I don't think Quiesce does that, it just flushes all the local data.

Wouldn't we would either have to use w:3 for all or a single extra w:3 after the LoadInitialData phases. Option 1 would be the easiest assuming it doesn't add too much overhead to the test runtime.

Reply:
The quiesce does do a "w:all". It figures out how many nodes there are and uses that in the write concern among other things. https://github.com/mongodb/genny/blob/master/src/gennylib/src/quiesce.cpp#L68-L80

Reply:
TIL, I thought that was just a fysnc, which would not have been sufficient. But seems we can keep this part as is :)

Reply:
I think I must have known that in the past but had forgotten so good to know!